.travis.yml

```diff
@@ -0,0 +1,35 @@
+language: java
+
+hosts:
+  - TRAVIS
+
+services:
+  - postgresql
+
+jdk:
+  - oraclejdk8
+
+install: mvn install -Dgpg.skip
+
+before_script:
+  - psql -c 'create database gemini_test;' -U postgres
+  - mvn archetype:update-local-catalog
+  - >
+      mvn archetype:generate -DarchetypeRepository=local -DarchetypeCatalog=local
+      -DarchetypeGroupId=com.techempower -DarchetypeArtifactId=gemini-resin-archetype
+      -Dpackage=foo.test -DartifactId=test-artifact-id -Dversion=1.0
+      -DmachineName=TRAVIS -DinteractiveMode=false
+      -DdatabaseHost=host.docker.internal
+      -DdatabaseName=gemini_test
+      -DdatabaseUsername=postgres
+  - cd test-artifact-id && mvn compile war:war
+  - curl -sL http://caucho.com/download/resin-4.0.56.tar.gz | tar xz --strip-components=1
+  - rm -rf webapps/*
+  - cp target/test-artifact-id-1.0.war webapps/ROOT.war
+
+script:
+  - java -jar lib/resin.jar console &
```

Comment:
This does not work as-is because the generated `web.xml` calls for the `default` servlet for static files (html/css/etc), which is created/named in our [`resin.xml`](https://github.com/TechEmpower/gemini/blob/master/gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/resin.xml#L15) (ignored at present).

The mechanism by which we run here is similar to how we do it inside a docker container - assume resin is installed/configured correctly by default (a decent assumption, in my opinion) and simply drops in our war as the default. In order to make this work, we have to change our `web.xml` to a slimmer and more out-the-box-compliant version:

```
<?xml version="1.0" encoding="UTF-8"?>
<web-app>
  <multipart-form enable="true" />

  <character-encoding>UTF-8</character-encoding>

  <servlet>
    <servlet-name>Server</servlet-name>
    <servlet-class>${package}.Server</servlet-class>
    <load-on-startup/>
  </servlet>
  <servlet-mapping>
    <servlet-name>Server</servlet-name>
    <url-pattern>*</url-pattern>
  </servlet-mapping>
</web-app>
```